src/core.js

```diff
@@ -425,6 +425,16 @@ jQuery.extend({
 	support: support
 });
 
+// jsHint would error on this code due to the Symbol not being defined in ES5.
+// Defining this global in .jshintrc would create a danger of using the global
+// unguarded in another place, it seems safer to just disable jsHint for these
+// three lines.
+/* jshint ignore: start */
+if ( typeof Symbol === "function" && Symbol.iterator ) {
```

Comment:
@rwaldron: I was responding to both you and @ljharb. I acknowledge that `for…of` is not the only use of iterators, and advocate for letting jQuery be exactly as iterable as arrays sharing its environment (and hence not to check for `Symbol.iterator` at all).